server/asset/eth/common.go

```diff
@@ -57,32 +76,49 @@ func (ss SwapState) String() string {
 	return "unknown"
 }
 
-// DecodeCoinID decodes the coin ID into flags, a contract address, and secret hash.
-func DecodeCoinID(coinID []byte) (uint16, common.Address, []byte, error) {
+// DecodeCoinID decodes the coin ID into flags, a contract address which may be
+// zeroed, and hash that represents either a secret or txid depending on flags.
+func DecodeCoinID(coinID []byte) (CoinIDFlag, *common.Address, []byte, error) {
+	var addr common.Address
 	if len(coinID) != coinIDSize {
-		return 0, common.Address{}, nil, fmt.Errorf("coin ID wrong length. expected %d, got %d",
+		return 0, nil, nil, fmt.Errorf("coin ID wrong length. expected %d, got %d",
 			coinIDSize, len(coinID))
 	}
-	secretHash := make([]byte, 32)
-	copy(secretHash, coinID[22:])
-	return binary.BigEndian.Uint16(coinID[:2]), common.BytesToAddress(coinID[2:22]), secretHash, nil
+	hash := make([]byte, 32)
+	copy(hash, coinID[22:])
+	copy(addr[:], coinID[2:22])
+	return CoinIDFlag(binary.BigEndian.Uint16(coinID[:2])),
+		&addr, hash, nil
 }
 
 // CoinIDToString converts coinID into a human readable string.
 func CoinIDToString(coinID []byte) (string, error) {
-	flags, addr, secretHash, err := DecodeCoinID(coinID)
+	flags, addr, hash, err := DecodeCoinID(coinID)
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("%x:%x:%x", flags, addr, secretHash), nil
+	return fmt.Sprintf("%x:%x:%x", flags, addr, hash), nil
 }
 
-// ToCoinID converts the address and secret hash to a coin ID.
-func ToCoinID(flags uint16, addr *common.Address, secretHash []byte) []byte {
+// ToCoinID converts the address and secret hash, or txid to a coin ID.
```

Comment:
Let's add that the `hash []byte` should be 32 bytes, and longer slices are truncated.  Either that or we panic or something, but a doc would be fine imo.

Reply:
Added a doc.